### PR TITLE
Build with OpenSSL 3 [skip ci]

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Build with OpenSSL 3.

### CODE REVIEW

File: abs.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+channels:
```
Comment (boldorider4) [inline abs.yaml]: Don't forget to remove this before merging!

**Review by boldorider4 — APPROVED** [review]

**Review by danpetry — COMMENTED** [review]
I know it's just a rebuild, but IMO it would be good to clean the tests up. As the linter says, the tests in the recipe won't be run, just the ones in the run_test.sh

**Comment by JeanChristopheMorinPerso** [comment]
Thanks @danpetry . I addressed your comment.

**Review by danpetry — APPROVED** [review]
Thanks! I would say the tests are better out of the recipe and in a run_test.sh, though. Tests in a recipe are usually pretty brief. Sorry should have been clearer. Approving based on these changes being addressed.

**Comment by JeanChristopheMorinPerso** [comment]
@danpetry I agree. It's what I wanted to do initially but I dislike the duplication required. So in this case I felt like the recipe tests were better than test scripts.

**Comment by danpetry** [comment]
@JeanChristopheMorinPerso sorry which duplication do you mean? AFAIU the commands in the `test` section are just bash/cmd commands anyway

**Comment by JeanChristopheMorinPerso** [comment]
>which duplication do you mean? AFAIU the commands in the test section are just bash/cmd commands anyway

Exactly that, the commands have to be duplicated for windows and unix. In this case it's the for loops that would have to be duplicated.

**Comment by danpetry** [comment]
ah right, fair enough. Yeah it's quite usual to have pretty similar `run_test.sh`/`run_test.bat` alongside each other (or also `build.sh`/`bld.bat`)

**Comment by danpetry** [comment]
Upon checking, it's not very common at all... I was thinking more about build.sh and bld.bat...

File: recipe/meta.yaml [hunk 2]
```diff
@@ -42,9 +44,13 @@ requirements:
...
-    - zlib
-    - libcurl  # [not win]
-    - openssl  # [not win]
+    - zlib {{ zlib }}
+    - libcurl 7.88.1  # [not win]
+    - openssl {{ openssl }}  # [not win]
+    # Pin to 11.2.0 until we figure out why we get "Error: Kind 16 not supported for type REAL at (1)"
+    - libgfortran5 11.2.0  # [unix]
+  run:
+    - openssl 3.*
```
Comment (danpetry) [inline recipe/meta.yaml]: I think this should have `# [not win]` after it since you want to ignore_run_export for `openssl` on `win`

**Review by danpetry — APPROVED** [review]
all good, issues with consuming package solved

**Pull request merged**